src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,200 @@
+                + " Join manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(selectRequest)) {
+            return getListByResultSet(statement.executeQuery());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all car", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET manufacturer_id = ?, model = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateRequest)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            deleteDrivers(car);
+            insertDrivers(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update this car - " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String updateRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateRequest)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by this id - " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectRequest = "SELECT c.id AS car_id, model, name, country, manufacturer_id"
+                + " FROM cars c "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = false AND driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(selectRequest)) {
+            statement.setLong(1, driverId);
+            return getListByResultSet(statement.executeQuery());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars by this driver id - "
+                    + driverId, e);
+        }
+    }
+
+    private void insertDrivers(Car car) {
+        String insertRequest = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
are u sure that you need  Statement.RETURN_GENERATED_KEYS?

Reply:
I think yes, because it's insert request